### PR TITLE
Add test case for `AbstractShenyuWasmPlugin`

### PR DESCRIPTION
**Pull request by Divyansh200102**
#5427

Make sure that:

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Comment by Divyansh200102** [comment]
@loongs-zhang I am getting a null pointer exception here could you please take a look?

image

```powershell
WARNING: If a serviceability tool is not in use, please run with -Djdk.instrument.traceUsage for more information
WARNING: Dynamic loading of agents will be disallowed by default in a future release
Java HotSpot(TM) 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
18:03:17.704 [main] DEBUG io.github.kawamuray.wasmtime.NativeLibraryLoader - Loading Wasmtime JNI library from C:\Users\Divyansh\AppData\Local\Temp\wasmtime_jni_0.19.0_windows_x86_648134173620995520118.dll
18:03:17.760 [main] DEBUG io.github.kawamuray.wasmtime.Func - New trampoline 0 of type FuncType(params=[I64, I64, I32], results=[I32])
18:03:17.763 [main] DEBUG io.github.kawamuray.wasmtime.Func - New trampoline 1 of type FuncType(params=[I64, I64, I32], results=[I32])
18:03:18.280 [main] DEBUG reactor.util.Loggers - Using Slf4j logging framework
rust side-> buffer base address: 1048496
18:03:19.532 [main] DEBUG io.github.kawamuray.wasmtime.Func - Trampoline 0 invoked with params=[Val(type=I64, value=0), Val(type=I64, value=1048496), Val(type=I32, value=32)], results=[null]
18:03:19.533 [main] INFO org.apache.shenyu.plugin.wasm.base.AbstractShenyuWasmPlugin - java side->hello from java 0
rust side-> recv:hello from java 0
18:03:19.539 [main] DEBUG io.github.kawamuray.wasmtime.Func - Trampoline 1 invoked with params=[Val(type=I64, value=0), Val(type=I64, value=1048780), Val(type=I32, value=11)], results=[null]
18:03:19.539 [main] INFO org.apache.shenyu.plugin.wasm.base.AbstractShenyuWasmPlugin - java side->rust result
18:03:19.544 [main] ERROR org.apache.shenyu.plugin.wasm.base.AbstractShenyuWasmPlugin - doExecute function not found in org.apache.shenyu.plugin.wasm.base.AbstractShenyuWasmPluginTest$RustShenyuWasmPlugin.wasm

java.lang.NullPointerException: Cannot invoke "org.apache.shenyu.plugin.api.result.ShenyuResult.error(org.springframework.web.server.ServerWebExchange, int, String, Object)" because the return value of "org.apache.shenyu.plugin.api.result.ShenyuResultWrap.shenyuResult()" is null

```

File: shenyu-plugin/shenyu-plugin-api/src/main/java/org/apache/shenyu/plugin/api/ShenyuPluginChain.java [hunk 1]
```diff
@@ -32,4 +34,5 @@ public interface ShenyuPluginChain {
      * @return {@code Mono<Void>} to indicate when request handling is complete
      */
     Mono<Void> execute(ServerWebExchange exchange);
+    Mono<Void> doExecute(ServerWebExchange exchange, SelectorData selector, RuleData rule);
```
Comment (loongs-zhang) [inline shenyu-plugin/shenyu-plugin-api/src/main/java/org/apache/shenyu/plugin/api/ShenyuPluginChain.java:37]: This class should not be modified.

File: shenyu-plugin/shenyu-plugin-wasm-base/pom.xml [hunk 2]
```diff
@@ -46,5 +46,10 @@
             <artifactId>jackson-module-kotlin</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.shenyu</groupId>
+            <artifactId>shenyu-plugin-wasm-base</artifactId>
+            <version>${project.version}</version>
+        </dependency>
```
Comment (loongs-zhang) [inline shenyu-plugin/shenyu-plugin-wasm-base/pom.xml:53]: rollback this

**Comment by loongs-zhang** [comment]
It seems that you did not regenerate the wasm file, please refer to https://github.com/apache/shenyu/blob/master/shenyu-plugin/shenyu-plugin-wasm-api/src/test/README.md

**Comment by Divyansh200102** [comment]
> It seems that you did not regenerate the wasm file, please refer to https://github.com/apache/shenyu/blob/master/shenyu-plugin/shenyu-plugin-wasm-api/src/test/README.md

I actually did generate the wasm file and according to the documentation I put it in the resources folder

image

image

**Comment by loongs-zhang** [comment]
see https://github.com/apache/shenyu/pull/5450

**Pull request closed without merging**